packages/arb-bridge-peripherals/contracts/tokenbridge/libraries/gateway/GatewayRouter.sol

```diff
@@ -143,6 +154,73 @@ abstract contract GatewayRouter is TokenGateway {
             );
     }
 
+    /**
+     * @notice Bridge ERC20 token using the registered or otherwise default gateway with call to permit.
+     * @param _token L1 address of ERC20
+     * @param _refundTo Account, or its L2 alias if it have code in L1, to be credited with excess gas refund in L2
+     * @param _to Account to be credited with the tokens in the L2 (can be the user's L2 account or a contract), not subject to L2 aliasing
+                  This account, or its L2 alias if it have code in L1, will also be able to cancel the retryable ticket and receive callvalue refund
+     * @param _amount Token Amount
+     * @param _maxGas Max gas deducted from user's L2 balance to cover L2 execution
+     * @param _gasPriceBid Gas price for L2 execution
+     * @param _data encoded data from router and user
+     * @param _permitData signature and deadline params of permit
+     * @param _isStandardImpl if true, standard implementation of permit; if false, could be similar to Dai impl
+     * @return res abi encoded inbox sequence number
+    */
+    function outboundTransferWithPermit(
+        address _token,
+        address _refundTo,
+        address _to,
+        uint256 _amount,
+        uint256 _maxGas,
+        uint256 _gasPriceBid,
+        bytes calldata _data,
+        PermitData calldata _permitData,
+        bool _isStandardImpl
+    ) public payable virtual returns (bytes memory) {
+        address gateway = getGateway(_token);
+        bytes memory gatewayData = GatewayMessageHandler.encodeFromRouterToGateway(
+            msg.sender,
+            _data
+        );
+
+        emit TransferRouted(_token, msg.sender, _to, gateway);
+        if (_isStandardImpl) {
+            ERC20Permit(_token).permit(
+                msg.sender,
+                gateway,
+                _amount,
+                _permitData.deadline,
+                _permitData.v,
+                _permitData.r,
+                _permitData.s
+            );
+        } else {
+            DaiToken(_token).permit(
+                msg.sender,
+                gateway,
+                _permitData.nonce,
+                _permitData.deadline,
+                true,
+                _permitData.v,
+                _permitData.r,
+                _permitData.s
+            );
+        }
+
+        return
+            ITokenGateway(gateway).outboundTransferCustomRefund{ value: msg.value }(
```

Comment:
one problem is that some legacy gateway might not have the `outboundTransferCustomRefund` method and will revert, we might need another entrypoint

Reply:
do you think we should have 2 functions in the router then, outboundTransferWithPermit and outboundTransferCustomRefundWithPermit?